components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/editor/js/design-view/cron-generator/cron-generator.js

```diff
@@ -0,0 +1,771 @@
+                    var firstIndexValue = dayOfWeek.charAt(0);
+                    var secondIndexValue = dayOfWeek.charAt(1);
+                    if(dayOfWeek.length === 1){
+                        return isValidateNumericValue([dayOfWeek], 1, 7)
+                    } else {
+                        return (!isNaN(parseInt(firstIndexValue)) && !isNaN(parseInt(secondIndexValue))) ?
+                                isValidateNumericValue([dayOfWeek], 1, 7) : isValidateStringValue([dayOfWeek], WEEK_ARRRAY);
+                    }
+                } else {
+                    return false;
+                }
+            });
+        };
+
+        var isValidCronExpression = function(cronExpression) {
+            if(!/\s/g.test(cronExpression)) {
+                return false;
+            }
+            var cronArray = cronExpression.split(" ");
+            if(parseInt(cronArray.length) !== 6) {
+                return false;
+            }
+            var seconds = cronArray[0];
+            var minutes = cronArray[1];
+            var hours = cronArray[2];
+            var dayOfMonth = cronArray[3];
+            var month = cronArray[4];
+            var dayOfWeek = cronArray[5];
+
+            var isValidSeconds = isValidTimeValue(seconds, 59);
+            var isValidMinutes = isValidTimeValue(minutes, 59);
+            var isValidHour = isValidTimeValue(hours, 23);
+            var isValidDayOfMonth = isValidDayOfMonthValue(dayOfMonth, dayOfWeek);
+            var isValidMonth = isValidMonthValue(month);
+            var isValidDayOfWeek = isValidDayOfWeekValue(dayOfWeek, dayOfMonth);
+
+            var isValidCron = isValidSeconds && isValidMinutes && isValidHour && isValidDayOfMonth &&
+                                                                            isValidMonth && isValidDayOfWeek;
+
+            if(isValidCron) {
+                return isValidCron;
+            }
+        };
+
+        CronGenerator.prototype.initialization = function(optionParent){
+            var self = this;
+            var generate = $('<div id = "generate" class = "col-xs-2" style="margin:10px 5px;">'+
+                    '<button type = "button" class = "btn btn-primary" '+
+                    'id = "generate-btn"> Schedule </button></div>');
+            $(optionParent).find('.option-content').append(generate);
+            $(optionParent).find('.option-value').attr('readonly', true);
+            $(optionParent).find('.option-value').after(' <i'+
+                ' class="fw fw-info"><span id="crondescription" style = "display:none"></span></i>');
+            var expression = $(optionParent).find('.option-value').val();
+            var description = Cronstrue.toString(expression, { throwExceptionOnParseError: false });
+            $('#source-options #crondescription').text(description);
+            $('#source-options #generate-btn').on('click',function(){
+                 new CronGenerator().render(optionParent);
```

Comment:
why you need to create a new object ???

Reply:
I need to clone while opening the modal dialog box frequently

Reply:
this is a hacky way of coding . You have already created a cron generator object when the attributes for file is generated. So in here you only need to do this call the render funtion like self.render(). So you render funtion need to change to create the popup from scratch everytime user clicks generate

Reply:
fixed in 0fb87f3